app/main.py

```diff
@@ -2,24 +2,71 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
+        self.name = name
+
+    def __get__(self, instance: str, owner: str) -> None:
+        return getattr(instance, self.name)
```

Comment:
The `__get__` method should return the value from the instance's dictionary using `instance.__dict__[self.name]` instead of `getattr(instance, self.name)`. This is because descriptors are meant to manage instance attributes directly.